crates/example-types/src/block_types.rs

```diff
@@ -365,3 +376,24 @@ impl Committable for TestBlockHeader {
         "TEST_HEADER".to_string()
     }
 }
+
+#[async_trait]
+impl TestableDelay for TestBlockHeader {
+    async fn handle_async_delay(delay_config: DelayConfig) {
+        if let Some(settings) = delay_config.get_setting(SupportedTypes::BlockHeader) {
+            match settings.delay_option {
+                DelayOptions::None => {}
+                DelayOptions::Fixed => {
+                    async_sleep(Duration::from_millis(settings.fixed_time_in_milliseconds)).await;
+                }
+                DelayOptions::Random => {
+                    let sleep_in_millis = rand::thread_rng().gen_range(
+                        settings.min_time_in_milliseconds..=settings.max_time_in_milliseconds,
+                    );
+
+                    async_sleep(Duration::from_millis(sleep_in_millis)).await;
+                }
+            }
+        }
+    }
+}
```

Comment:
It looks like most of this code is duplicated for the various types. If it's the same thing, I think `handle_async_delay` could have a default implementation in the trait, and you could have each type just implement e.g. `settings(DelayConfig) -> Option<DelaySettings>`

Reply:
yes it is duplicated, i thought the same thing, but left it like this incase in the future we have more specific tests

Reply:
thanks for pointing this out! i realized the code was duplicated but this default method will help clean it up!